### ISSUE TITLE
Difficulty getting accurate Nav state after js navigation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [cloudtrends/chromiumembedded#213]
```
What steps will reproduce the problem?
1. Using cefclient go to http://twitter.com/#search?q=%23dontactlikeyounever
2. On the left bar click on #bornthisday
3. Click the back button

What is the expected output? What do you see instead?
After step 3. I expect the Forward button to be available. It is not available.

What version of the product are you using? On what operating system?
r207, windows 7 64bit


What I'm running into is that after using a js navigation like the one above 
HandleLoadEnd is not called, so the nav state isn't updated. I could check the 
nav state after the HandleAddressChange (which is fired) however WebHistory 
isn't update until after that handler call.

I'd like to either update the WebHistory before HandleAddressChange, or add a 
new handler function HandleNavState(bool can_go_back, bool can_go_forward) that 
is fired when CefBrowserImpl::set_nave_state is called. The first is less 
obtrusive but could change how people expect the history to work. I can create 
a patch on this, just looking for some direction on which would be better (or 
if you have an alternate solution)
```

Original issue reported on code.google.com by `ddmac...@gmail.com` on 28 Mar 2011 at 9:33
- Merged into: #208
